src/components/NewCommentForm.tsx

```diff
@@ -2,7 +1,87 @@
+import cl from 'classnames';
+import React, { FormEvent, useState } from 'react';
+import { addComment } from '../api/api';
+import { pause } from '../utils/methods';
+import { Comment } from '../types/Comment';
+
+interface Props {
+  postId: number;
+  onAddComment: (v: Comment) => void;
+}
+
+export const NewCommentForm: React.FC<Props> = ({ postId, onAddComment }) => {
+  const [nameError, setNameError] = useState(false);
+  const [emailError, setEmailError] = useState(false);
+  const [bodyError, setBodyError] = useState(false);
+
+  const [name, setName] = useState('');
+  const [email, setEmail] = useState('');
+  const [body, setBody] = useState('');
+
+  const [isLoading, setIsLoading] = useState(false);
+
+  const handleAddChange = async () => {
+    setNameError(!name);
+    setEmailError(!email);
+    setBodyError(!body);
+
+    if (!name || !email || !body) {
+      return;
+    }
+
+    const newComment = {
+      postId,
+      name,
+      email,
+      body,
+    };
+
+    setIsLoading(true);
+    try {
+      await pause();
+
+      const res = await addComment(newComment);
+
+      onAddComment(res);
+      setBody('');
+    } catch {
+      throw new Error('Unable to add a comment');
```

Comment:
Throwing an error here is not user-friendly. Consider displaying an error message to the user instead of throwing an error.